frappe/model/base_document.py

```diff
@@ -391,10 +400,11 @@ def db_insert(self):
 		columns = list(d)
 		try:
 			frappe.db.sql("""INSERT INTO `tab{doctype}` ({columns})
-					VALUES ({values})""".format(
-					doctype = self.doctype,
-					columns = ", ".join("`"+c+"`" for c in columns),
-					values = ", ".join(["%s"] * len(columns))
+					VALUES ({values}) {conflict_handler}""".format(
+					doctype=self.doctype,
+					columns=", ".join("`"+c+"`" for c in columns),
+					values=", ".join(["%s"] * len(columns)),
+					conflict_handler=conflict_handler
```

Comment:
Reminds me of this mess 🥲 => https://github.com/frappe/frappe/pull/14197/files#diff-e6f487d771b5a26f0c26d0716fb3f33ec71538c5492f181656e79c5e06aacd47R71-R77